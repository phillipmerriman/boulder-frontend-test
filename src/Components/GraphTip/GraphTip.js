import React from 'react';

const GraphTip = () => {
  return <h3 data-testid='axisInfo'>X-Axis: Milliseconds | Y-Axis: Power Output</h3>;
};

export default GraphTip;
