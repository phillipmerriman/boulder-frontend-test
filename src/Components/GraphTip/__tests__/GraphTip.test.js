import React from 'react';
import ReactDOM from 'react-dom';
import GraphTip from '../GraphTip';
import { render, cleanup, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

afterEach(cleanup);

it('renders without crashing', () => {
    const div = document.createElement("div");
    ReactDOM.render(<GraphTip />, div);
});

it('renders axis info correctly', () => {
    render(<GraphTip />);
    expect(screen.getByTestId('axisInfo')).toHaveTextContent('X-Axis: Milliseconds | Y-Axis: Power Output');
});

it('matches snapshot', () => {
    const tree = renderer.create(<GraphTip />).toJSON();
    expect(tree).toMatchSnapshot();
});
