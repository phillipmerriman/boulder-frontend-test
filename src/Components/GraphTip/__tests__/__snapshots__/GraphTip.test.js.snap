// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphTip component matches snapshot 1`] = `
<h3
  data-testid="axisInfo"
>
  X-Axis: Milliseconds | Y-Axis: Power Output
</h3>
`;
